src/vs/workbench/contrib/chat/common/chatEditingService.ts

```diff
@@ -110,19 +117,100 @@ export const enum ChatEditingSessionChangeType {
+	readonly resource: URI;
+	readonly snapshotUri: URI;
+	readonly state: WorkingSetEntryState;
+	telemetryInfo: IModifiedEntryTelemetryInfo;
+}
+
+export interface ITextSnapshotEntry extends IBaseSnapshotEntry {
+	kind: 'text';
+	readonly languageId: string;
+	readonly original: string;
+	readonly current: string;
+	readonly originalToCurrentEdit: OffsetEdit;
+	serialize(): Promise<ITextSnapshotEntryDTO>;
+}
+
+export interface INotebookSnapshotEntry extends IBaseSnapshotEntry {
+	kind: 'notebook';
+	serialize(): Promise<INotebookSnapshotEntryDTO>;
+}
+export type ISnapshotEntry = ITextSnapshotEntry | INotebookSnapshotEntry;
+
+export interface IBaseSnapshotEntryDTO {
+	readonly resource: string;
+	readonly state: WorkingSetEntryState;
+	readonly snapshotUri: string;
+	readonly telemetryInfo: IModifiedEntryTelemetryInfoDTO;
+}
+export interface ITextSnapshotEntryDTO extends IBaseSnapshotEntryDTO {
+	readonly kind: 'text';
+	readonly languageId: string;
+	readonly originalHash: string;
+	readonly currentHash: string;
+	readonly originalToCurrentEdit: IOffsetEdit;
+}
+
+export interface INotebookSnapshotEntryDTO extends IBaseSnapshotEntryDTO {
+	readonly kind: 'notebook';
+}
+export type ISnapshotEntryDTO = ITextSnapshotEntryDTO | INotebookSnapshotEntryDTO;
+
+interface IModifiedEntryTelemetryInfoDTO {
+	readonly requestId: string;
+	readonly agentId?: string;
+	readonly command?: string;
+}
+
+
+export interface IModifiedEntryTelemetryInfo {
+	readonly agentId: string | undefined;
+	readonly command: string | undefined;
+	readonly sessionId: string;
+	readonly requestId: string;
+	readonly result: IChatAgentResult | undefined;
+}
+
+interface IModifiedAnyFileEntry extends IDisposable {
+	readonly entryId: string;
 	readonly originalURI: URI;
-	readonly originalModel: ITextModel;
 	readonly modifiedURI: URI;
+	readonly onDidDelete: Event<void>;
 	readonly state: IObservable<WorkingSetEntryState>;
 	readonly isCurrentlyBeingModified: IObservable<boolean>;
 	readonly rewriteRatio: IObservable<number>;
 	readonly diffInfo: IObservable<IDocumentDiff>;
 	readonly lastModifyingRequestId: string;
+	readonly telemetryInfo: IModifiedEntryTelemetryInfo;
 	accept(transaction: ITransaction | undefined): Promise<void>;
 	reject(transaction: ITransaction | undefined): Promise<void>;
+	acceptAgentEdits(textEdits: TextEdit[], isLastEdits: boolean): void;
+	acceptStreamingEditsStart(tx: ITransaction): void;
+	acceptStreamingEditsEnd(tx: ITransaction): void;
+	updateTelemetryInfo(telemetryInfo: IModifiedEntryTelemetryInfo): void;
+	resetToInitialValue(): void;
 }
 
+export interface IModifiedTextFileEntry extends IModifiedAnyFileEntry {
+	readonly kind: 'text';
+	readonly originalModel: ITextModel;
+	readonly modifiedModel: ITextModel;
+	createSnapshot(requestId: string | undefined): ITextSnapshotEntry;
+	restoreFromSnapshot(snapshot: ITextSnapshotEntry): void;
+}
+
+export interface IModifiedNotebookFileEntry extends IModifiedAnyFileEntry {
```

Comment:
Separation of IModifiedFileEntry into text and notebook.
Similarly the snapshot interfaces have been separated, giving each their own implementations